### PR TITLE
fix(forms/datalist): merge async titlemaps in schema

### PR DESCRIPTION
**Pull request by ncomont**
**What is the problem this PR is trying to solve?**

When issued from a trigger, the datalist titleMap is not merged in its schema. In a ComponentForm context, the name resolution is not possible (`$XXX_name` is not sent).

**What is the chosen solution to this problem?**

🙃 

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: output/forms.eslint.txt [hunk 1]
```diff
@@ -5,6 +5,9 @@
 /home/travis/build/Talend/ui/packages/forms/src/UIForm/UIForm.container.js
   94:2  error  componentWillReceiveProps is deprecated since React 16.9.0, use UNSAFE_componentWillReceiveProps instead, see https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops. Use https://github.com/reactjs/react-codemod#rename-unsafe-lifecycles to automatically update your components  react/no-deprecated
 
+/home/travis/build/Talend/ui/packages/forms/src/UIForm/fields/Datalist/Datalist.component.js
+  92:2  error  hasTitleMap should be placed after getTitleMap  react/sort-comp
```
Comment (ncomont) [inline output/forms.eslint.txt:9]: 😒

File: packages/forms/src/UIForm/fields/Datalist/Datalist.component.test.js [hunk 2]
```diff
@@ -154,6 +154,12 @@ describe('Datalist component', () => {
 					...schema.schema,
 					enum: ['my', 'title', 'map'],
 				},
+
+				titleMap: [
+					{ name: 'my_name', value: 'my' },
+					{ name: 'title_name', value: 'title' },
+					{ name: 'map_name', value: 'map' },
+				],
```
Comment (jsomsanith-tlnd) [inline packages/forms/src/UIForm/fields/Datalist/Datalist.component.test.js:162]: This doesn't change anything in your test.
The scenario you describe should be tested: 
- no titlemap
- trigger to get the titlemap
- check that on change, titlemap is present

File: packages/forms/src/UIForm/fields/Datalist/Datalist.component.js [hunk 3]
```diff
@@ -62,7 +63,18 @@ class Datalist extends Component {
...
+			...payload,
+			schema: { ...mergedSchema },
+		};
+
+		if (this.hasTitleMap()) {
+			payloadWithSchema = {
+				...payloadWithSchema,
+				schema: { ...payloadWithSchema.schema, titleMap: this.getTitleMap() },
+			};
+		}
```
Comment (jsomsanith-tlnd) [inline packages/forms/src/UIForm/fields/Datalist/Datalist.component.js:76]: (Small suggestion that doesn't really change anything, so if you disagree, it's ok)
What should be tested is that if there is a titlemap in state (so from trigger), so we add it.
```suggestion
		const payloadWithSchema = {
			...payload,
			schema: { ...mergedSchema, titleMap: this.state.titleMap || mergedSchema.titleMap },
		};
```

File: output/forms.eslint.txt [hunk 4]
```diff
@@ -5,6 +5,9 @@
 /home/travis/build/Talend/ui/packages/forms/src/UIForm/UIForm.container.js
   94:2  error  componentWillReceiveProps is deprecated since React 16.9.0, use UNSAFE_componentWillReceiveProps instead, see https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops. Use https://github.com/reactjs/react-codemod#rename-unsafe-lifecycles to automatically update your components  react/no-deprecated
 
+/home/travis/build/Talend/ui/packages/forms/src/UIForm/fields/Datalist/Datalist.component.test.js
+  253:9  error  'wrapper' was used before it was defined  @typescript-eslint/no-use-before-define
+
 /home/travis/build/Talend/ui/packages/forms/src/UIForm/fields/Enumeration/EnumerationWidget.js
```
Comment (VolodymyrKovalM) [inline output/forms.eslint.txt:11]: please fix error if possible

**Pull request merged**